src/fluree/server/consensus/standalone.clj

```diff
@@ -0,0 +1,135 @@
+
+(defn broadcast-new-ledger!
+  "Responsible for producing the event broadcast to connected peers."
+  [subscriptions watcher {:keys [ledger-id server tx-id commit-file-meta] :as handler-result}]
+  (log/info (str "New Ledger successfully created by server " server ": " ledger-id " with tx-id: " tx-id "."))
+  (watcher/deliver-watch watcher tx-id handler-result)
+  (subscriptions/send-message-to-all subscriptions "ledger-created" ledger-id (:json commit-file-meta))
+  :success)
+
+(defn parse-opts
+  "Extract the opts from the transaction and keywordify the top level keys."
+  [expanded-txn]
+  (let [string-opts (get-first-value expanded-txn const/iri-opts)]
+    (reduce-kv (fn [opts k v]
+                 (assoc opts (keyword k) v))
+               {} string-opts)))
+
+(defn create-ledger!
+  [conn subscriptions watcher {:keys [ledger-id txn opts tx-id] :as _params}]
+  (go-try
+    (let [create-opts    (parse-opts txn)
+          ledger         (deref!
+                          (fluree/create conn ledger-id create-opts))
+          staged-db      (-> ledger
+                             fluree/db
+                             (fluree/stage txn opts)
+                             deref!)
+          commmit-result (deref!
+                         ;; following uses :file-data? and will return map with {:keys [db data-file commit-file]}
+                          (fluree/commit! ledger staged-db {:file-data? true}))]
+      (broadcast-new-ledger! subscriptions watcher (assoc commmit-result :tx-id tx-id
+                                                          :ledger-id ledger-id
+                                                          :t 1)))))
+
+(defn broadcast-new-commit!
+  "Responsible for producing the event broadcast to connected peers."
+  [subscriptions watcher {:keys [ledger-id tx-id server commit-file-meta] :as commit-result}]
+  (log/info "New transaction completed for" ledger-id "tx-id: " tx-id "by server:" server)
+  (watcher/deliver-watch watcher tx-id commit-result)
+  (subscriptions/send-message-to-all subscriptions "new-commit" ledger-id (:json commit-file-meta))
+  :success)
+
+(defn transact!
+  [conn subscriptions watcher {:keys [ledger-id tx-id txn opts] :as params}]
+  (go-try
+    (let [start-time    (System/currentTimeMillis)
+          _             (log/debug "Starting transaction processing for ledger:" ledger-id
+                                   "with tx-id" tx-id ". Transaction sat in queue for"
+                                   (- start-time (:instant params)) "milliseconds.")
+          ledger        (if (deref! (fluree/exists? conn ledger-id))
+                          (deref! (fluree/load conn ledger-id))
+                          (throw (ex-info "Ledger does not exist" {:ledger ledger-id})))
+          staged-db     (-> ledger
+                            fluree/db
+                            (fluree/stage txn opts)
+                            deref!)
+          commit-result (deref!
```

Comment:
Per the `process-event` exception handling comment below, here we'll throw and never make it to delivering a message that the HTTP handler is waiting for.

Probably here we don't throw, then deliver the response below (which might be an exception).

Same thing with `create-ledger!` above.

Reply:
I think I've covered this scenario by [reusing the error handling machinery from the raft implementation](https://github.com/fluree/server/pull/75/files#diff-f93dffc5ef2fc673ab168a1b72e7c8d268aeea3ffb8e92c9760bb23da3384997R75-R78). If there's an error while processing events, then that error is broadcasted out and the assigned watcher is delivered with the error.